front/src/modules/simulationResult/helpers/makeDraggingState.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  isPointPickingElement,
+  isSegmentPickingElement,
+  type HoveredItem,
+} from '@osrd-project/ui-charts';
+
+import {
+  extractPacedTrainIdFromOccurrenceId,
+  isOccurrenceId,
+  isOccurrenceProjection,
+  isTrainId,
+} from 'utils/trainId';
+
+import type { IndividualTrainProjection, TrainSpaceTimeData } from '../types';
+
+export default function makeDraggingState(
+  hoveredItem: HoveredItem,
+  projectedTrains: IndividualTrainProjection[],
+  projectPathTrainResults: TrainSpaceTimeData[]
+): {
+  draggedTrain: IndividualTrainProjection;
+  initialDepartureTime: Date;
+} | null {
+  if (
+    !isSegmentPickingElement(hoveredItem.element) &&
+    !isPointPickingElement(hoveredItem.element)
+  ) {
+    throw new Error('wrong hoveredItem type');
+  }
+
+  const hoveredTrainId = hoveredItem.element.pathId;
+  if (!isTrainId(hoveredTrainId)) {
+    throw new Error('hovered train id should be a train schedule id or an occurrence id');
+  }
+  const train = projectedTrains.find((projectedTrain) => projectedTrain.id === hoveredTrainId);
+
+  if (!train) {
+    console.error(`No train found with id ${hoveredTrainId}`);
+    return null;
+  }
+  if (isOccurrenceId(hoveredTrainId)) {
+    const occurrenceProjection = projectedTrains.find(
+      ({ id }) => isOccurrenceId(id) && hoveredTrainId === id
+    )!;
+    if (
+      occurrenceProjection &&
+      isOccurrenceProjection(occurrenceProjection) &&
+      !!occurrenceProjection.isStartTimeException
```

Comment:
It's already a boolean
```suggestion
      occurrenceProjection.isStartTimeException
```

Reply:
done